activesupport/test/cache/behaviors/cache_store_behavior.rb

```diff
@@ -498,7 +498,41 @@ def test_cache_miss_instrumentation
     ActiveSupport::Notifications.unsubscribe "cache_read.active_support"
   end
 
+  def test_nil_is_significant
+    skip "does not support blank key" if does_not_support_blank_keys?
+
+    @cache.write(nil, "baz")
+    assert_equal "baz", @cache.read(nil)
+
+    @cache.write([nil], "bizz")
+    assert_equal "bizz", @cache.read(nil)
+  end
+
+  def test_nil_is_significant_in_array_and_hash
+    @cache.write([ :a, nil, :b, "", :c ], "foo")
+    assert_equal "foo", @cache.read("a//b//c")
+
+    @cache.write({ hello: nil, world: "today", nil: "rules" }, "bar")
+    assert_equal "bar", @cache.read("hello=/nil=rules/world=today")
+  end
+
+  def test_empty_string_behavior
+    @cache.write(["", "", ""], "foo")
+    assert_equal "foo", @cache.read("//")
```

Comment:
Can cover empty array elements in the array test above. At this level, we're getting close to indirectly testing to_param behavior. We could do detailed unit tests, echoing those for to_param, on Key, leaving cache behavior tests free of details beyond "yes, Key handles serialization of key names."

Reply:
Sounds good. I'm going to move these remaining test cases directly to the Key class in my other PR:

```
  def test_nil_is_significant_in_array_and_hash
    @cache.write([ :a, nil, :b, "", :c ], "foo")
    assert_equal "foo", @cache.read("a//b//c")

    @cache.write({ hello: nil, world: "today", nil: "rules" }, "bar")
    assert_equal "bar", @cache.read("hello=/nil=rules/world=today")
  end

  def test_empty_string_behavior
    @cache.write(["", "", ""], "foo")
    assert_equal "foo", @cache.read("//")
  end
```